modules/Lending.py

```diff
@@ -98,6 +102,21 @@ def notify_summary(sleep_time):
     scheduler.enter(sleep_time, 1, notify_summary, (sleep_time, ))
 
 
+def notify_new_loans(sleep_time):
+    global loans_provided
+    new_provided = api.return_active_loans()['provided']
+    if loans_provided:
+        get_ids = lambda loans: map(lambda l: l['id'], loans)
+        new_loan_ids = set(get_ids(new_provided)) - set(get_ids(loans_provided))
+        for id in new_loan_ids:
+            loan = filter(lambda l: l['id'] == id, new_provided)[0]
```

Comment:
List comprehensions ftw! I'd take both those changes, I'm just used to using callbacks and nodejs.

> One thing though, what if there's more than one loan in your 60s sleep period?

That's what the loop is for. The filter just gets the list of all loans with a specific ID, which should always have a single element.